var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -30,30 +30,35 @@ class PyScipy(PythonPackage):
         'scipy.sparse.linalg.eigen.lobpcg', 'scipy.special._precompute'
     ]
 
-    version('1.3.1', sha256='2643cfb46d97b7797d1dbdb6f3c23fe3402904e3c90e6facfe6a9b98d808c1b5')
-    version('1.3.0', sha256='c3bb4bd2aca82fb498247deeac12265921fe231502a6bc6edea3ee7fe6c40a7a')
-    version('1.2.2', sha256='a4331e0b8dab1ff75d2c67b5158a8bb9a83c799d7140094dda936d876c7cfbb1')
-    version('1.2.1', sha256='e085d1babcb419bbe58e2e805ac61924dac4ca45a07c9fa081144739e500aa3c')
-    version('1.1.0', sha256='878352408424dffaa695ffedf2f9f92844e116686923ed9aa8626fc30d32cfd1')
-    version('1.0.0', sha256='87ea1f11a0e9ec08c264dc64551d501fa307289460705f6fccd84cbfc7926d10')
+    version('1.4.1',  sha256='dee1bbf3a6c8f73b6b218cb28eed8dd13347ea2f87d572ce19b289d6fd3fbc59')
+    version('1.4.0',  sha256='31f7cfa93b01507c935c12b535e24812594002a02a56803d7cd063e9920d25e8')
+    version('1.3.3',  sha256='64bf4e8ae0db2d42b58477817f648d81e77f0b381d0ea4427385bba3f959380a')
+    version('1.3.2',  sha256='a03939b431994289f39373c57bbe452974a7da724ae7f9620a1beee575434da4')
+    version('1.3.1',  sha256='2643cfb46d97b7797d1dbdb6f3c23fe3402904e3c90e6facfe6a9b98d808c1b5')
+    version('1.3.0',  sha256='c3bb4bd2aca82fb498247deeac12265921fe231502a6bc6edea3ee7fe6c40a7a')
+    version('1.2.3',  sha256='ecbe6413ca90b8e19f8475bfa303ac001e81b04ec600d17fa7f816271f7cca57')
+    version('1.2.2',  sha256='a4331e0b8dab1ff75d2c67b5158a8bb9a83c799d7140094dda936d876c7cfbb1')
+    version('1.2.1',  sha256='e085d1babcb419bbe58e2e805ac61924dac4ca45a07c9fa081144739e500aa3c')
+    version('1.1.0',  sha256='878352408424dffaa695ffedf2f9f92844e116686923ed9aa8626fc30d32cfd1')
+    version('1.0.0',  sha256='87ea1f11a0e9ec08c264dc64551d501fa307289460705f6fccd84cbfc7926d10')
     version('0.19.1', sha256='a19a2ca7a7336495ec180adeaa0dfdcf41e96dbbee90d51c3ed828ba570884e6')
-    version('0.19.0', sha256='4190d34bf9a09626cd42100bbb12e3d96b2daf1a8a3244e991263eb693732122',
-            url="https://pypi.io/packages/source/s/scipy/scipy-0.19.0.zip")
     version('0.18.1', sha256='8ab6e9c808bf2fb3e8576cd8cf07226d9cdc18b012c06d9708429a821ac6634e')
     version('0.17.0', sha256='f600b755fb69437d0f70361f9e560ab4d304b1b66987ed5a28bdd9dd7793e089')
     version('0.15.1', sha256='a212cbc3b79e9a563aa45fc5c517b3499198bd7eb7e7be1e047568a5f48c259a')
     version('0.15.0', sha256='0c74e31e08acc8bf9b6ceb9bced73df2ae0cc76003e0366350bc7b26292bf8b1')
 
-    depends_on('python@2.6:2.8,3.2:')
-    depends_on('python@2.7:2.8,3.4:', when='@0.18:')
-    depends_on('python@3.5:', when='@1.3:')
+    depends_on('python@2.6:2.8,3.2:', type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.4:', when='@0.18:', type=('build', 'run'))
+    depends_on('python@3.5:', when='@1.3:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
-    depends_on('py-pytest', type='test')
+    depends_on('py-pybind11@2.4.0:', when='@1.4.1:')
+    depends_on('py-pybind11@2.2.4:', when='@1.4.0:')
```

Comment:
@matz-e : FYI, in the upstream we had:

```
-    depends_on('py-pybind11@2.4.0:', when='@1.4.1:', type='build')
-    depends_on('py-pybind11@2.2.4:', when='@1.4.0:', type='build')
```

Note that people often put `build` dependency assuming package is used during building only. This is incorrect and it should be also `link` dependency. IIRC, such things I saw in some bbp packages as well.

Reply:
This is a Python package though, which should not really be linked further?

Reply:
I did saw compilation/link error and that's when noticed this dependency. (don't want to guess what tensorflow does with it's dependencies :) )

Reply:
…sigh…